src/daemon/daemon.cpp

```diff
@@ -786,10 +786,14 @@ std::string generate_unused_mac_address(std::unordered_set<std::string>& s)
 {
     // TODO: Checking in our list of MAC addresses does not suffice to conclude the generated MAC is unique. We
     // should also check in the ARP table.
-    while (true)
+    static constexpr auto max_tries = 5;
+    for (auto i = 0; i < max_tries; ++i)
         if (auto [it, success] = s.insert(mp::utils::generate_mac_address()); success)
             return *it;
 
+    throw std::runtime_error{
+        fmt::format("Failed to generate an unique mac address after {} attempts. Number of mac addresses in use: {}",
+                    max_tries, s.size())};
```

Comment:
Given that the MAC generator always generate addresses started in `52:54:00:`, we have 16 million possible MAC addresses. For this, I think we'll never generate even one repeated MAC. I think that error would indicate something very weird is happening. I'm ok with the change, although I don't foresee a possibility to fail.

Reply:
Exactly. I agree with what you say and that was exactly why I chose to add it with a relatively small number of retries. Chance of collision is very low if everything is correct, so we never expect 5 straight collisions. That would signal problems that would be difficult to understand otherwise.

Some thoughts to explain my reasoning. With a uniform mac distribution, if we have `n` allocated macs, the probability would be `(n/16^6)^5`. With `n=100` allocated macs, that would be on the order of `10^-27`. However, a small bug can radically change that. Suppose we have some sort of leak such that `n` reaches 10 million: the probability is now ~8% (that's why I included the number of used macs in the exception msg). Or suppose the generation is very biased and hits the same macs for some reason. Without this, it could just take longer to generate, which would be difficult to identify/understand.

Strictly speaking, exceptions are meant to deal with unexpected conditions, not to signal bugs - asserts should be used for that. However, we could conceivably hit an external condition causing repetition (like something wrong with the source of entropy in the OS or something). So, when I wrote the `while(true)` I got goosebumps and thought it was better this way.

Reply:
I agree.